### PR TITLE
enhancement: directly compare with TimeSystem conditions

### PR DESCRIPTION
**Pull request by WeylonSantana**
Before it was necessary to define a variable for the value "Time system" so that we could compare with our variables that use the time system, this modification allows to compare directly.

https://user-images.githubusercontent.com/63019821/215845980-fc58147b-0e5a-474a-a2d1-b097e22af4f3.mp4

### CODE REVIEW

**Comment by Arufonsu** [comment]
Could you please show how it looks like when you add/remove this variable comparison/condition (?) also, can you use the former english editor for this (?) ( despite the fact i know spanish, this is kinda confusing to me tbh ). 

Anyways, seeing that this adds up a new way to compare with the system time directly **but** this is indirectly a feature that already exists if certain steps are followed, i'll tag it as _minor feature addition or quality of life change_ (Enhancement).

**Comment by lodicolo** [comment]
> Could you please show how it looks like when you add/remove this variable comparison/condition (?)

I would also like to see what this actually looks like when adding/removing the condition.

**Comment by WeylonSantana** [comment]
Idk if you want this screenshoot precisely, but

before:
image

after:
image
image

**Comment by lodicolo** [comment]
> (@lodicolo, @Arufonsu) Idk if you want this screenshoot precisely, but

Yes, and I am considering that we begin requiring screenshots and/or video clips to make doc writing easier.

File: Intersect.Editor/Forms/Editors/Events/Event Commands/EventCommand_ConditionalBranch.cs [hunk 1]
```diff
@@ -866,6 +871,10 @@ private void TryLoadVariableIntegerComparison(VariableCompaison comparison)
                     cmbCompareUserVar.SelectedIndex = UserVariableBase.ListIndex(integerComparison.CompareVariableId);
                 }
             }
+            else if(integerComparison.TimeSystem == true)
```
Comment (lodicolo) [inline Intersect.Editor/Forms/Editors/Events/Event Commands/EventCommand_ConditionalBranch.cs:874]: Drop ` == true`, this isn't a nullable

File: Intersect.Editor/Localization/Strings.cs [hunk 2]
```diff
@@ -419,6 +419,11 @@ public static string GetVariableComparisonString(IntegerVariableComparison compa
             if (comparison.CompareVariableId == Guid.Empty)
             {
                 value = comparison.Value.ToString();
+
+                if(comparison.TimeSystem == true)
```
Comment (lodicolo) [inline Intersect.Editor/Localization/Strings.cs:423]: `if (`, and drop ` == true`

File: Intersect.Editor/Forms/Editors/Events/Event Commands/EventCommand_ConditionalBranch.cs (hunk 1 above)
Comment (lodicolo) [reply Intersect.Editor/Forms/Editors/Events/Event Commands/EventCommand_ConditionalBranch.cs:874]: Also `if (` (add a space)

File: Intersect.Editor/Localization/Strings.cs [hunk 3]
```diff
@@ -2661,6 +2666,8 @@ public partial struct EventConditionDesc
 
             public static LocalizedString UserVariable = @"{00}: {01} {02}";
 
+            public static LocalizedString SystemTime = @"System Time (ms)";
```
Comment (lodicolo) [inline Intersect.Editor/Localization/Strings.cs:2669]: Can you move this up to L2659 to sort it?

File: Intersect.Server/Entities/Events/Conditions.cs [hunk 4]
```diff
@@ -614,6 +614,11 @@ Event instance
                     compValue = player.User.GetVariableValue(comparison.CompareVariableId);
                 }
             }
+            else if(comparison.TimeSystem == true)
```
Comment (lodicolo) [inline Intersect.Server/Entities/Events/Conditions.cs:617]: `if (`, drop ` == true`

File: Intersect.Server/Entities/Events/Conditions.cs [hunk 5]
```diff
@@ -614,6 +614,11 @@ Event instance
...
+            {
+                compValue = (long)(DateTime.UtcNow - new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc))
```
Comment (lodicolo) [inline Intersect.Server/Entities/Events/Conditions.cs:619]: Is there any way we can just not have to subtract Jan 1 1970?

**Review by lodicolo — CHANGES_REQUESTED** [review]

File: Intersect.Server/Entities/Events/Conditions.cs (hunk 5 above)
Comment (WeylonSantana) [reply Intersect.Server/Entities/Events/Conditions.cs:619]: https://dirask.com/posts/C-NET-get-current-time-in-milliseconds-gp5BRj

**Review by lodicolo — APPROVED** [review]

**Pull request merged**